pkg/storer/internal/chunkstore/transaction.go

```diff
@@ -165,16 +173,27 @@ func (cs *txChunkStoreWrapper) NewTx(state *storage.TxState) storage.TxChunkStor
 		toReleaseLocs: make(map[[32]byte]sharky.Location),
 		toReleaseSums: make(map[sharky.Location][32]byte),
 	}
+	chunkStore := &chunkStoreWrapper{
+		store:  cs.store,
+		sharky: txSharky,
+	}
 	return &txChunkStoreWrapper{
 		TxChunkStoreBase: &storage.TxChunkStoreBase{
-			TxState: state,
-			ChunkStore: &chunkStoreWrapper{
-				store:  cs.store,
-				sharky: txSharky,
-			},
+			TxState:    state,
+			ChunkStore: chunkStore,
 		},
 		store:    cs.store,
 		txSharky: txSharky,
+		revOps: storage.NewInMemTxRevertOpStore(
+			map[storage.TxOpCode]storage.TxRevertFn[swarm.Address, swarm.Chunk]{
+				storage.PutOp: func(key swarm.Address, val swarm.Chunk) error {
+					return chunkStore.Delete(context.Background(), key)
+				},
+				storage.DeleteOp: func(key swarm.Address, val swarm.Chunk) error {
+					return chunkStore.Put(context.Background(), val)
+				},
+			},
+		),
 	}
 }
 
```

Comment:
Can we remove the txSharky allocation in NewTxChunkStore? These are never used.

Reply:
Done.